### PR TITLE
Closes #634: nginx configuration for staging and production

### LINKED ISSUE #634: nginx configuration for staging and production
**What would you like to be added**:

A few things we should do in our nginx config for staging and production:

- setup nginx to [serve our static content](https://docs.nginx.com/nginx/admin-guide/web-server/serving-static-content/) (e.g., GatsbyJS frontend) instead of the node app
- setup proper [caching for our endpoints](https://serversforhackers.com/c/nginx-caching) to reduce load on the node app

**Why would you like this to be added**:

The nginx server is better a certain tasks than our node app, and we should leverage it where possible.

### PR DESCRIPTION
**Pull request by manekenpix**
## Issue This PR Addresses

Closes #634 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

This adds using nginx to serve Telescope's static content and caching Telescope's endpoints to reduce load on it.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -1,6 +1,14 @@
 events {}
 
 http {
+  # levels        - Defines hierarchy levels
+  # keys_zone     - Name for these settings
+  # inactive      - Cached data that are not accessed during the time specified by the inactive parameter get removed from the cache regardless of their freshness
+  # max_size      - When this size is exceeded, it removes the least recently used data
+  # use_temp_path - Use a temp directory before moving all cached files into the cache directory
```
Comment (humphd) [inline nginx.conf]: Thanks for these comments, helpful.

**Comment by manekenpix** [comment]
I'm not sure, I think we'll probably have to do it here since `nginx.conf` has this:
```
# Disables processing of certain response header fields from the proxied server
proxy_ignore_headers "Cache-Control" "Expires";
```

**Comment by manekenpix** [comment]
@c3ho what do you think?

**Comment by humphd** [comment]
From https://docs.nginx.com/nginx/admin-guide/content-cache/content-caching/#select

>To define conditions under which NGINX Plus does not send cached responses to clients, include the proxy_cache_bypass directive. Each parameter defines a condition and consists of a number of variables. If at least one parameter is not empty and does not equal “0” (zero), NGINX Plus does not look up the response in the cache, but instead forwards the request to the backend server immediately.

```
proxy_cache_bypass $cookie_nocache $arg_nocache$arg_comment;
```

>To define conditions under which NGINX Plus does not cache a response at all, include the proxy_no_cache directive, defining parameters in the same way as for the proxy_cache_bypass directive.

```
proxy_no_cache $http_pragma $http_authorization;
```

**Comment by manekenpix** [comment]
@humphd I made changes to avoid caching `/admin`, `/user` and `/health`. Can you review these changes?

File: nginx.conf [hunk 2]
```diff
@@ -17,12 +25,45 @@ http {
...
     ssl_certificate_key /etc/letsencrypt/live/dev.telescope.cdot.systems/privkey.pem;
 
-    location / {
+   # Let's Encrypt route
+    location /.well-known/acme-challenge/ {
+      root /var/www/certbot;
+    }
+
+    # Non-cached routes /admin /user /health
+    location ~ ^/(admin|user|health) {
```
Comment (humphd) [inline nginx.conf]: Let's add `/auth` to this list too

Comment (manekenpix) [reply nginx.conf]: Added `/auth` and comments in `index.js`.

File: src/backend/web/routes/index.js [hunk 3]
```diff
@@ -17,11 +17,19 @@ const query = require('./query');
 
 const router = express.Router();
 
+/**
+ * In staging and production, our revers proxy takes care of serving the content in the public folder.
```
Comment (humphd) [inline src/backend/web/routes/index.js]: "reverse"

File: src/backend/web/routes/index.js [hunk 4]
```diff
@@ -17,11 +17,19 @@ const query = require('./query');
...
+ * We're keeping this route for development.
+ */
 router.use(express.static(path.join(__dirname, '../../../frontend/public')));
```
Comment (humphd) [inline src/backend/web/routes/index.js]: If this is only for development, should we wrap this in a `if(process.NODE_ENV !== 'production')` or something?

Comment (manekenpix) [reply src/backend/web/routes/index.js]: I honestly don't know. In production and staging, this route will never be used, so I'm open to suggestions.

Comment (humphd) [reply src/backend/web/routes/index.js]: In `env.staging` we have `NODE_ENV=production`, and `env.example` has `NODE_ENV=development` so let's do this:

```js
if(process.NODE_ENV === 'devleopment') {
  router.use(express.static(path.join(__dirname, '../../../frontend/public')));
}
```

**Comment by raygervais** [comment]
Tested locally, this is awesome!

**Pull request merged**